Sources/Parsing/Parsers/Bool.swift

```diff
@@ -19,24 +19,34 @@ extension Bool {
     .init()
   }
 
-  /// A parser that consumes a Boolean value from the beginning of a substring.
+  /// A parser that consumes a Boolean value from the beginning of a substring's UTF-8 view.
   ///
-  /// This parser only recognizes the literal "true" and "false" characters:
+  /// This overload is provided to allow the `Input` generic to be inferred when it is
+  /// `Substring.UTF8View`.
   ///
-  /// ```swift
-  /// Bool.parser().parse("true Hello"[...])  // (output: true, rest: " Hello")
-  /// Bool.parser().parse("false Hello"[...]) // (output: false, rest: " Hello")
-  /// Bool.parser().parse("1 Hello"[...])     // (output: nil, rest: "1 Hello")
-  /// ```
+  /// - Parameter inputType: The `Substring.UTF8View` type. This parameter is included to mirror the
+  ///   interface that parses any collection of UTF-8 code units.
+  /// - Returns: A parser that consumes a Boolean value from the beginning of a substring's UTF-8
+  ///   view.
+  @inlinable
+  public static func parser(
+    of inputType: Substring.UTF8View.Type = Substring.UTF8View.self
+  ) -> Parsers.BoolParser<Substring.UTF8View> {
+    .init()
+  }
```

Comment:
An example of a new overload that favors `UTF8View` when encountering a `UInt8` byte. Cuts down on things like `Prefix<Substring.UTF8View>`, where `Prefix` will now do.